src/DependencyGraph.java

```diff
@@ -0,0 +1,169 @@
+     * DependencyVertex. The secondary is removed as a subdependency of the primary
+     * @param dep
+     * @param subDep
+     */
+    void removeSubDependency(DependencyVertex dep, DependencyVertex subDep) {
+        deps.get(dep).remove(subDep);
+    }
+
+    /**
+     * This method takes in a DependencyVertex and removes it from all
+     * subdependency lists
+     * @param subDep
+     */
+    void removeSubDependency(DependencyVertex subDep) {
+        deps.values().removeAll(Collections.singleton(subDep));
+    }
+
+    /**
+     * This method returns a boolean indicating whether or not the provided
+     * DependencyVertex already exists in the graph
+     * @param dep
+     * @return
+     */
+    public boolean hasDependency(DependencyVertex dep) {
+        return deps.containsKey(dep);
+    }
+
+    /**
+     * This method wth produce a result indicating whether or not a dependency relationship
+     * exists between the two dependencies
+     * @param dep1 assumed as the parent dependency
+     * @param dep2 assumed as the sub dependency
+     * @return 1 if dep2 is a subdependency of dep1, -1 if dep1 is a subdependency of dep2,
+     * 0 if neither are related
+     */
+    public int hasInterdependency(DependencyVertex dep1, DependencyVertex dep2) {
+        if(deps.get(dep1).contains(dep2)) {
+            return 1;
+        }
+
+        if(deps.get(dep2).contains(dep1)) {
+            return -1;
+        }
+
+        return 0;
+    }
+
+    /**
+     * This method returns the number of dependencies in the graph
+     * @return number of dependencies
+     */
+    public int getDependencyCount() {
+        return deps.keySet().size();
+    }
+
+    /**
+     * This method returns the number of subdependencies in the graph
+     * @return number of subdependencies
+     */
+    public int getInterdependenciesCount() {
+        int count = 0;
+        for (DependencyVertex v : deps.keySet()) {
+            count += deps.get(v).size();
+        }
+        return count;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder builder = new StringBuilder();
+
+        for(DependencyVertex  v: deps.keySet()) {
+            builder.append(v.toString() + " \033[1;37m" + "requires" + "\033[0m { ");
+            for(DependencyVertex w : deps.get(v)) {
+                builder.append(w.toString() + " , ");
+            }
+            builder.append(" }\n");
+        }
+
+        return (builder.toString());
+    }
+}
+
+class DependencyVertex {
+    String name;
+    String groupID;
+    String version;
+
+    DependencyVertex(String name, String groupID, String version) {
+        this.name = name;
+        this.groupID = groupID;
+        this.version = version;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null) {
+            return false;
+        }
+
+        if (getClass() != other.getClass()) {
```

Comment:
This auto generated code is accounting for possible subclasses. It's a lot simpler (in general) to just make the class final. You also get it for free with a record